gamemode-news-hook wrong announcement.json link

### Describe the bug

Checking the logs I'm getting this error.
After investigating the GitHub link should be "stable" instead of "main".
```
aug 21 18:00:36 bazzite gamemode-news-hook[2039]: [INFO] Injection complete, monitoring...
aug 21 18:00:36 bazzite gamemode-news-hook[2039]: [INFO] Observer result: {"marker":"<200b>","version":"1.4.0","patched":0}
aug 21 18:00:36 bazzite gamemode-news-hook[2039]: [INFO] XHR hook result (BP): {"error":"failed to fetch our events"}
aug 21 18:00:36 bazzite gamemode-news-hook[2039]: [INFO] Injecting XHR hook + MutationObserver into BigPicture...
aug 21 18:00:36 bazzite gamemode-news-hook[2039]: [INFO] XHR hook result (SJC): {"error":"failed to fetch our events"}
aug 21 18:00:36 bazzite gamemode-news-hook[2039]: [INFO] Injecting XHR hook into SharedJSContext...
aug 21 18:00:36 bazzite gamemode-news-hook[2039]: [ERROR] All repo mirrors failed
aug 21 18:00:36 bazzite gamemode-news-hook[2039]: [WARNING] Mirror failed https://raw.githubusercontent.com/ublue-os/bazzite-gamemode-news/refs/heads/main/announcements.json: HTTP 404
```

### What did you expect to happen?

To fetch the announcements.json without error.

### Session Logs

```shell
https://paste.centos.org/view/0fc69e04
```

### Hardware

Custom PC: Intel i7-7700K, Radeon RX 6650 XT, 16GB DDR4, HDMI>Sony TV.

### Extra information or context

_No response_

--- a/Containerfile
+++ b/Containerfile
@@ -777,7 +777,12 @@ RUN --mount=type=cache,dst=/var/cache \
         dnf5 -y copr disable -y $copr; \
     done && unset -v copr && \
     { rm -v /usr/share/applications/bazzite-steam-bpm.desktop || true; } && \
-    sed -i "s|^github = .*|github = https://raw.githubusercontent.com/ublue-os/bazzite-gamemode-news/refs/heads/${IMAGE_BRANCH}/announcements.json|" /etc/gamemode-news-hook.conf && \
+    news_branch="${IMAGE_BRANCH}" && \
+    case "${news_branch}" in \
+        testing|unstable) ;; \
+        *) news_branch="stable" ;; \
+    esac && \
+    sed -i "s|^github = .*|github = https://raw.githubusercontent.com/ublue-os/bazzite-gamemode-news/refs/heads/${news_branch}/announcements.json|" /etc/gamemode-news-hook.conf && \
     mkdir -p /usr/lib/systemd/user/gamescope-session-plus@ogui-steam.service.wants && \
     ln -s /usr/lib/systemd/user/steamos-powerbuttond.service /usr/lib/systemd/user/gamescope-session-plus@ogui-steam.service.wants/ && \
     sed -i 's/@steam/@ogui-steam/g' /usr/lib/systemd/user/gamemode-news-hook.service && \

--- a/tests/dgoss/tests.d/00-smoke/goss.yaml
+++ b/tests/dgoss/tests.d/00-smoke/goss.yaml
@@ -13,6 +13,29 @@ command:
       '
     exit-status: 0
 
+  gamemode_news_hook_uses_release_channel:
+    exec: |
+      sh -lc '
+      config=/etc/gamemode-news-hook.conf
+      if [ ! -f "$config" ]; then
+        echo "SKIP: gamemode-news-hook is not installed"
+        exit 0
+      fi
+      url=$(sed -n "s/^github[[:space:]]*=[[:space:]]*//p" "$config")
+      case "$url" in
+        */refs/heads/stable/announcements.json|*/refs/heads/testing/announcements.json|*/refs/heads/unstable/announcements.json)
+          echo "OK: gamemode-news-hook uses a release channel ($url)"
+          ;;
+        *)
+          echo "FAIL: gamemode-news-hook uses a non-release announcement URL: ${url:-<missing>}"
+          exit 1
+          ;;
+      esac
+      '
+    exit-status: 0
+    stdout:
+      - "/^(OK|SKIP):/"
+
   selinux_label_probe_works:
     exec: |
       sh -lc '

